src/core/Akka.Remote/Serialization/DaemonMsgCreateSerializer.cs

```diff
@@ -114,6 +114,17 @@ public override object FromBinary(byte[] bytes, Type type)
         {
             var proto = DaemonMsgCreateData.ParseFrom(bytes);
             var clazz = Type.GetType(proto.Props.Clazz);
+
+            if (clazz == null)
+            {
+                var msg =
+                    string.Format(
+                        "Could not find '{0}' in loaded assemblies. If this is a remote/cluster deploy scenario, " +
+                        "ensure the remote system has referenced the assembly containing this type.", proto.Props.Clazz);
+                    
+                throw new ActorInitializationException(msg);
```

Comment:
Yes, you're right, we shouldnt really clutter things up by introducing more and more exception types.
The TypeLoadException is the thing that one should look for in this scenario I guess, so lets just go with `Type.GetType(...,true)` then.